Adjust docs to vouchers improvements (#1000)
saleor/saleor-docs: docs/developer/discounts/vouchers.mdx

## Introduction

Vouchers are a way to apply discounts to orders. They can be used to reduce the cost of a checkout.

> Vouchers can be applied to the entire order or to specific products. They can also be limited to a minimum quantity of products in the checkout.

Reviewer: Maybe we should also mention here that the vouchers can be mixed with promotions?

To apply the voucher on checkout use [checkoutAddPromoCode](api-reference/checkout/mutations/checkout-add-promo-code.mdx)
mutation. The discount will be visible both in the line prices and in the `checkout.discount` field. Let's see the examples.

### Usage limits

There are multiple ways to limit a voucher's usage, but default voucher codes can be used without any limits. To change that `usageLimit` can be set
to limit the number of times a voucher can be used. The voucher will be unavailable for further use when the limit is reached.
If the voucher has multiple codes, the limit is calculated as the sum of all codes usage.

> Another way to limit the usage of a voucher is to set the `usageLimitPerCustomer` field to `true`. In this case, the voucher code will be

Reviewer 2: The flag is named `applyOncePerCustomer` and not `usageLimitPerCustomer`.

available for use only once per customer.

The voucher can be also set as single-use, when `singleUse` flag is set to `true`. In this case, each code from the voucher can be used only once and then the code will be deactivated after use.
**This flag can be updated on existing voucher only when no code has been used yet.**

> Those limitation options can be combined. For example, if `usageLimit` is set to `10` and `usageLimitPerCustomer` is set to `true`, the voucher can be used by the first 10 users.

Reviewer 2: The flag is named `applyOncePerCustomer` and not `usageLimitPerCustomer`.

To apply the voucher on checkout use [checkoutAddPromoCode](api-reference/checkout/mutations/checkout-add-promo-code.mdx)
mutation. The discount will be visible both in the line prices and in the `checkout.discount` field. Let's see the examples.

### Usage limits

There are multiple ways to limit a voucher's usage, but default voucher codes can be used without any limits. To change that `usageLimit` can be set
to limit the number of times a voucher can be used. The voucher will be unavailable for further use when the limit is reached.
If the voucher has multiple codes, the limit is calculated as the sum of all codes usage.

Another way to limit the usage of a voucher is to set the `usageLimitPerCustomer` field to `true`. In this case, the voucher code will be
available for use only once per customer.

> The voucher can be also set as single-use, when `singleUse` flag is set to `true`. In this case, the codes from the voucher can be used only once and then the code will be deactivated.
> 

Reviewer: Maybe
```suggestion
The voucher can be also set as single-use, when `singleUse` flag is set to `true`. In this case, each code from the voucher can be used only once and the code will be deactivated after use. 
```

            "hasPreviousPage": false
          }
        },
        "discountValueType": "FIXED",
        "startDate": "2023-10-23T08:20:12.215018+00:00",
        "endDate": null,
        "applyOncePerOrder": true,
        "applyOncePerCustomer": true,
        "singleUse": false
      }
    }
  }
}
```

### Updating voucher

To update a voucher use [voucherUpdate](api-reference/discounts/mutations/voucher-update.mdx) mutation.

Using this mutation voucher codes can be only added. To remove codes use [voucherDelete](api-reference/discounts/mutations/voucher-code-bulk-delete.mdx) mutation.

Input:

```json {3}
{
  "input": {
    "addCodes": ["NewCode"]
  }
}
```

Mutation:

```graphql {16-29}
mutation voucherUpdate($input: VoucherInput!) {
  voucherUpdate(input: $input) {
    errors {
      field
      code
      message
      voucherCodes
    }
    voucher {
      id
      type
      minCheckoutItemsQuantity
      name
      usageLimit
      used
      codes(first: 10) {
        edges {
          node {
            code
            used
          }
        }
        pageInfo {
          startCursor
          endCursor
          hasNextPage
          hasPreviousPage

>         }

Reviewer: Same here

      applyOncePerCustomer
      singleUse
    }
  }
}
```

Here is the response:

```json {12-39}
{
  "data": {
    "voucherUpdate": {
      "errors": [],
      "voucher": {
        "id": "Vm91Y2hlcjo0",
        "type": "ENTIRE_ORDER",
        "minCheckoutItemsQuantity": 10,
        "name": "NewVoucher",
        "usageLimit": 10,
        "used": 0,
        "codes": {
          "edges": [
            {
              "node": {
                "code": "NewCode",
                "used": 0
              }
            },
            {
              "node": {
                "code": "code3",
                "used": 0
              }
            },
            {
              "node": {
                "code": "code2",
                "used": 0
              }
            },
            {
              "node": {
                "code": "code1",
                "used": 0
              }
            }
          ],
          "pageInfo": {
            "startCursor": "WyIyMDIzLTEwLTIzIDA4OjIwOjEyLjIyNjQxMCswMDowMCIsICJjb2RlMyJd",
            "endCursor": "WyIyMDIzLTEwLTIzIDA4OjIwOjEyLjIyNjM4MyswMDowMCIsICJjb2RlMSJd",
            "hasNextPage": false,
            "hasPreviousPage": false

>           }

Reviewer: And here

The discount amount can be found in `order.discounts.amount`, the value is equal to the
`order.lines.unitDiscount` multiplied by the line quantity.
The discount can also be seen in the line prices - compare the `undiscountedUnitPrice` and the `unitPrice`.

## Webhooks

All voucher actions also generate asynchronous webhooks that can be used for reacting
to the events emitted by Saleor, e.g. `VOUCHER_CREATED`, `VOUCHER_UPDATED`, `VOUCHER_DELETED`.
The [WebhookEventTypeAsyncEnum](/api-reference/webhooks/enums/webhook-event-type-async-enum.mdx#values)
represents the full list of asynchronous webhooks.

You can learn more about webhooks [here](developer/extending/webhooks/asynchronous-events.mdx).

## Exporting voucher codes

Voucher codes can be exported to a CSV or XLSX file.

> You can learn more about vouchers codes export [here](developer/export/export-voucher-codes.mdx).

Reviewer 3: ```suggestion
You can learn more about voucher codes export [here](developer/export/export-voucher-codes.mdx).
```
